esp/platform/espsecurecontext.hpp

```diff
@@ -72,6 +75,8 @@ interface IEspSecureContext : extends IInterface
     //           DeserializationResult (*pfn)(const char*, TValue&)
     template <typename TValue, typename TDefault = TValue, class TDeserializer = TokenDeserializer>
     bool getProp(int type, const char* name, TValue& value, const TDefault dflt = TDefault(), DeserializationResult* deserializationResult = NULL, TDeserializer& deserializer = TDeserializer());
+
+    virtual void initSessionData(const char* name, const char* value) = 0;
```

Comment:
virtual void setSessionBased(bool sessionBased) = 0;
    virtual const bool getSessionBased() = 0;
    virtual void setUserData(const char* data) = 0;
    virtual const StringBuffer& getUserData(StringBuffer& data) = 0;

Reply:
This is focused on session. Is session generic enough to exist in the base interface, or is it a specific to HTTP? If the concept of sessions is broader than HTTP(S), and the ESP is providing a single data value for manager use, this is fine.

I know scapps uses non-HTTP protocols, and I don't know if we'll need to support them. If sessions are HTTP specific, I would prefer pairing the generic getProp with a matching setProp.

Reply:
I am not sure we need to pair the generic getProp with a matching setProp.

Reply:
`IEspSecureContext` is intended to be protocol agnostic, in the event we need to support protocols other than HTTP/HTTPS. If sessions are specific to HTTP/HTTPS, then the interface should probably not add session-specific interfaces when they have no meaning to other protocols.

Only if not including session-specific interfaces because sessions are HTTP/HTTPS-specific would a generic `setProp` be needed.

Once again, are sessions HTTP/HTTPS specific?

Reply:
I may mis-understand the HTTP/HTTPS specific. The initial intention of adding the session is to replace the browser login. It is only for HTTP/HTTPS.

Reply:
To make the getSessionBased/setSessionBased more generic, we may rename them as getDomainAuthType/setDomainAuthType (for session based, the type could be 'Session'. Default could be 'Unknown').